### PR TITLE
Allow the indexing pipeline to report the indexing progress onto the UI

### PR DESCRIPTION
**Pull request by trducng**
This PR:
- Allows the indexing pipeline to show indexing progress.
- List out successful and failed files.

image

### CODE REVIEW

File: libs/ktem/ktem/index/file/pipelines.py [hunk 1]
```diff
@@ -236,226 +249,356 @@ def get_pipeline(cls, user_settings, index_settings, selected):
...
+                    ).all()
+                    for each in index:
+                        if each[0].relation_type == "vector":
+                            vs_ids.append(each[0].target_id)
+                        else:
+                            ds_ids.append(each[0].target_id)
+                        session.delete(each[0])
+                    session.commit()
+                self.VS.delete(vs_ids)
+                self.DS.delete(ds_ids)
```
Comment (lone17) [inline libs/ktem/ktem/index/file/pipelines.py]: this logic should be made into separate `delete_file()` function to simplify the logic of `run()` and avoid the duplication in `stream()`

File: libs/ktem/ktem/index/file/pipelines.py [hunk 2]
```diff
@@ -236,226 +249,356 @@ def get_pipeline(cls, user_settings, index_settings, selected):
...
```
Comment (lone17) [inline libs/ktem/ktem/index/file/pipelines.py]: same as above

**Review by lone17 — CHANGES_REQUESTED** [review]
Thanks @trducng. Just few small changes and I think it's good for a minor version pump.

File: libs/ktem/ktem/index/file/pipelines.py (hunk 2 above)
Comment (trducng) [reply libs/ktem/ktem/index/file/pipelines.py]: Changed in [bfff3c9](https://github.com/Cinnamon/kotaemon/pull/81/commits/bfff3c9a5aa1be55d4b79caff046a3f5858c9d3d)

File: libs/ktem/ktem/index/file/pipelines.py (hunk 1 above)
Comment (trducng) [reply libs/ktem/ktem/index/file/pipelines.py]: Changed in [bfff3c9](https://github.com/Cinnamon/kotaemon/pull/81/commits/bfff3c9a5aa1be55d4b79caff046a3f5858c9d3d)

**Review by lone17 — APPROVED** [review]
LGTM

**Pull request merged**